api/controllers/game/resolve.js

```diff
@@ -0,0 +1,435 @@
+module.exports = function (req, res) {
+
+  //Note: the player calling resolve is the opponent of the one playing the one-off, if it resolves
+  const promiseGame = gameService.findGame({gameId: req.session.game});
+  const promisePlayer = userService.findUser({userId: req.body.opId});
+  const promiseOpponent = userService.findUser({userId: req.session.usr});
+  const promisePlayerPoints = cardService.findPoints({userId: req.body.opId});
+  const promiseOpPoints = cardService.findPoints({userId: req.session.usr});
+  Promise.all([promiseGame, promisePlayer, promiseOpponent, promisePlayerPoints, promiseOpPoints])
+    .then(function changeAndSave(values) {
+      const [game, player, opponent, playerPoints, opPoints] = values;
+      let happened = true;
+      const playerUpdates = {};
+      const opponentUpdates = {};
+      let cardsToScrap = [];
+      let updatePromises = [];
+      let gameUpdates = {
+        oneOffTarget: null,
+        oneOffTargetType: '',
+      };
+      if (game.twos.length % 2 === 1) {
+        // One-off is countered
+        opponentUpdates.frozenId = null;
+        gameUpdates.log = [
+          ...game.log,
+          `The ${game.oneOff.name} is countered, and all cards played this turn are scrapped.`,
+        ];
+        happened = false;
+      } else {
+        playerUpdates.frozenId = null;
```

Comment:
Oops this will only clear the frozenId if the one-off resolves. We should:
1. add a test case to confirm the issue
2. Fix it by moving this line into the initial declaration of playerUpdates on line 13

Test case could be as follows: (from P1's pesprective)
1. P0 plays 9 on p1's whatever
2. P1 plays 6 to blow up face cards
3. P0 counters
4. P1 resolves (so 6 effect doesn't happen)
5. P1 plays their whatever again (it should no longer be frozen)

Reply:
Also I just realized that if the one-off is countered, it's the player themselves who resolves it, so in that case we need to set the opponent's frozenId to null. Which indicates the value of a also testing that frozenId doesn't incorrectly clear b/c my advice above was bad and would clear the wrong player's frozenId.

Instead of initializing `playerUpdates` to include the frozenId, we should initialize `playerUpdates` to an empt dictionary and set it when the one-off 'happened' (just like you're doing) but we should also initialize `opponentUpdates` to an empty dictionary, set its frozenId to null if the one-off **didn't happen**, and in the bottom of resolved when we're creating the `Promise.all()` with all the updates, also update the opponent in the same way we update the player, so as to clear whichever of their frozenIds we need to

Reply:
I now see that line 23 addresses this case. So with the test you added to confirm, I'm considering this ... resolved ;)

Reply:
So, added that test and it looks like it's not an issue (as it passes as it is currently).

I'll look into making the other changes you requested though